framework_tvb/tvb/core/services/crypto_service.py

```diff
@@ -0,0 +1,65 @@
+import os
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from cryptography.hazmat.backends import default_backend
+
+
+# noinspection PyTypeChecker
+class CryptoService(object):
+    _instance = None
+    KEY_FOLDER_NAME = 'PUBLIC_KEY'
+    KEY_FILE_NAME = 'public_key.pem'
+
+    @staticmethod
+    def decrypt_content(view_model, trait_upload_field_name):
+        if view_model.encrypted_aes_key is None:
+            return
+
+        # Open encrypted file
+        upload_path = getattr(view_model, trait_upload_field_name)
+        with open(upload_path, 'rb') as f:
+            encrypted_content = f.read()
+
+        # Get the encrypted AES symmetric-key
+        with open(view_model.encrypted_aes_key, 'rb') as f:
+            encrypted_aes_key = f.read()
+
+        # Read the private key
+        with open("../../adapters/uploaders/keys/private_key.pem", "rb") as key_file:
```

Comment:
I suggest to put the full path towards the Private Key inside .tvb.configuration (and then read it similarly to the way we read `KEYCLOAK_CONFIGURATION`)